actors/builtin/storage_market/storage_market_actor.go

```diff
@@ -339,31 +338,31 @@ func (a *StorageMarketActor) Constructor(rt Runtime) *vmr.EmptyReturn {
 
 func _rtAbortIfDealAlreadyProven(rt Runtime, deal OnChainDeal) {
 	if deal.SectorStartEpoch != epochUndefined {
-		rt.AbortStateMsg("Deal has already appeared in proven sector.")
+		rt.Abort(exitcode.ErrIllegalState, "Deal has already appeared in proven sector.")
```

Comment:
Not sure if these deal proposal related errors are IllegalState or IllegalArgument

Reply:
mmh argument is legal would be my case, but doesnt work coz of state

Reply:
Argument I think per re read